keras/src/quantizers/quantizers.py

```diff
@@ -127,6 +127,341 @@ def get_config(self):
         }
 
 
+def adjust_and_nudge(min_range, max_range, num_bits, narrow_range):
```

Comment:
Is `adjust_and_nudge` function following the implementation here?
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/fake_quant_ops_functor.h#L41
It seems slightly different.

Additionally, it might not work with `Model.fit` if the computation involves converting values to python `float` midway.